### DISCUSSION TITLE
Bulk actions with parameter

### DISCUSSION THREAD

**Discussion by eswachsman** [rappasoft/laravel-livewire-tables#253] [Feature Requests]
I might be missing something, but is there a way to execute a bulk action with a parameter. So, for example, I have an "assign to user" action and I want the user to be able to select which user to assign the records to.

Is there something like this already?

**Comment by rappasoft** [comment]
There is nothing like this currently, you just get an array of the primary keys of selected rows to work with.

**Reply by eswachsman** [reply]
Got it. Is this something you are thinking of adding to this awesome package?

**Reply by rappasoft** [reply]
Yeah, I have no problem adding it we just have to figure out the best way to do it.

**Comment by iAmKevinMcKee** [comment]
I would love to see this as well. My thought you could define array of options and then a message, and then have a modal pop up with a custom message.

```php
public array $bulkActions = [
    'exportSelected' => 'Export',
    ['assignToUser', 'Assign to User', 'assignToUserOptions', 'Please choose the user that these records should be assigned to.'],
];

public function assignToUser($userId)
{
    if($this->selectedRowsQuery->count()) {
        // map over selected rows and update the assigned user to the $userId
    }
}

public function assignToUserOptions(): array
{
    return User::pluck('name', 'id')->toArray();
}


```
@rappasoft if I made a PR for this, would you be willing to accept it?

**Reply by rappasoft** [reply]
Hmm, it would probably be more work than what you posted as I currently have no modal functionality built-in. If we were to figure that out, we would need to make it work for all templates. As well as have it be backward compatible. I'm definitely open to it as I've needed it myself, but it requires research.
